### ISSUE TITLE
Support for BunJS?

### ISSUE THREAD

**Issue by ArthurTimofey** [UrielCh/opencv4nodejs#153]
Hey,

I'm wondering if there is a potential way to support BunJS?

As for now I am running in to the following issue with regards to Node API:

```
https://github.com/oven-sh/bun/issues/158#issuecomment-2239879288
```

The API `napi_register_module_v1` is supported in BunJS but it seems like it still does not work for me. Been trying to hack it too with no progress but any suggestions very much welcome.

For reference I have also asked bun about this issue [here](https://github.com/oven-sh/bun/issues/158#issuecomment-2239879288).

**Comment by UrielCh** [comment]
Nope, this OpenCV4nodeJS is built on top of the previous Node native API. I tried to build a new one from scratch using the modern N-API and mapping the same binary as the official Python OpenCV API, but it took me more time than I can allocate to it.

See: https://github.com/UrielCh/opencv-sandbox

I have converted some common Python binding calls to N-API, and I think the project needs four more months to complete.

**Comment by ArthurTimofey** [comment]
Ah I see, well that's a shame but completely understandable.

Ill see if I can use the sandbox to try and use the functionality I need or even contribute to it.

Thanks for the great work anyways.

**Comment by UrielCh** [comment]
I finally change my mind and start to migrate the project to N-API so I will be able to use it with Deno and BunJS.

**Comment by stoefln** [comment]
@UrielCh how is it going? Is there something I can help with?

**Comment by UrielCh** [comment]
I have updated the project [node-addon-examples](https://github.com/nodejs/node-addon-examples/pulls) to improve the N-API samples (4 PR opened), but the maintainers don't seem very interested in my proposal.

The active branch for this migration is:
[napi-1](https://github.com/UrielCh/opencv4nodejs/tree/napi-1).
I have cleaned up the test folder to add features back one by one.

Currently, I need to focus on finding a full-time job, so I have little time to dedicate to this project.

**Comment by stoefln** [comment]
> I have updated the project [node-addon-examples](https://github.com/nodejs/node-addon-examples/pulls) to improve the N-API samples

Hmm. How is this related to opencv4nodejs n-api migration?
